packages/flutter/lib/src/material/button.dart

```diff
@@ -152,9 +154,30 @@ class MaterialButton extends StatefulWidget {
   /// The color to use for this button's text.
   final Color textColor;
 
-  /// The color of the button, as printed on the [Material].
+  /// The primary color of the button, as printed on the [Material], while it
+  /// is in its default (unpressed, enabled) state.
+  ///
+  /// Defaults to null, meaning that the color is automatically derived from the [Theme].
   final Color color;
 
+  /// The primary color of the button when the button is in the down state
+  /// (pressed) state. The splash is represented as a circular overlay that appears
+  /// above the [highlightColor] overlay. The splash overlay has a center point that
```

Comment:
I'm not convinced it'll always be over the highlight overlay. It probably depends on the exact timing. But it's probably good enough for the docs and we can fix the painting order later.